### ISSUE TITLE
(cli): `cdk --all` deploys hidden stage stacks not displayed with `cdk diff --all`

### ISSUE THREAD

**Issue by McDoit** [aws/aws-cdk#14898]
Running `cdk deploy --all` deploys stacks in stages, not visible when running `cdk diff --all`

Related to this feature: https://github.com/aws/aws-cdk/pull/14379

### Reproduction Steps

```
const app = new cdk.App()

// Stage allows us to synth CF template for Stacks added to it
const stage = new cdk.Stage(app, "Stage")
new StackToDeployAsStackset(stage, "XYZ")
const stackSetTemplateObj = stage.synth().stacks[0].template

const stackSetDeployer = new cdk.Stack(app, "StackSetDeployer")
new cdk.CfnStackSet(stackSetDeployer, "XYZASaStackSet", {
            ...
            templateBody: JSON.stringify(stackSetTemplateObj),
})
```

This creates both a stackset, but also the **hidden** stack deployment of **XYZ**

### What did you expect to happen?

Following the discussion about cdk and stacksets, creating a stage with a stack and synthing the stage to use that stack artifact with cloudformation stacksets, and with a stage, the stack shouldnt be deployed locally

Or at least visible in in the diff if this was intentional

### What actually happened?

The `deploy` command deploys the stack from the stage, and the `diff` command didn't even display that this would happen

### Environment

  - **CDK CLI Version  : 1.106.1**
  - **Framework Version: 1.106.1**
  - **Node.js Version: v15.8.0** 
  - **OS               : Windows**
  - **Language (Version): all** 

### Other

Related to this feature: https://github.com/aws/aws-cdk/pull/14379

--- 

This is :bug: Bug Report

**Comment by rix0rrr** [comment]
That should not happen, thanks for reporting.

**Comment by otaviomacedo** [comment]
Fixed by #15046.
